gctscript/modules/ta/indicators/indicators.go

```diff
@@ -0,0 +1,40 @@
+package indicators
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/gctscript/modules"
+)
+
+var supportedDurations = []string{"1m", "3m", "5m", "15m", "30m", "1h", "2h", "4h", "6h", "12h", "24h", "1d", "3d", "1w"}
+
+func toFloat64(data interface{}) (float64, error) {
+	switch d := data.(type) {
+	case float64:
+		return d, nil
+	case int:
+		return float64(d), nil
+	case int32:
+		return float64(d), nil
+	case int64:
+		return float64(d), nil
+	default:
+		return 0, fmt.Errorf(modules.ErrParameterConvertFailed, d)
+	}
+}
+
+func ParseInterval(in string) (time.Duration, error) {
+	if !common.StringDataContainsInsensitive(supportedDurations, in) {
+		return time.Nanosecond, errors.New("invalid interval")
+	}
+	switch in {
+	case "1d":
+		in = "24h"
+	case "3d":
+		in = "72h"
```

Comment:
You are changing a supported interval to an unsupported interval? In the case above of 1d, you have both 24h and 1d in the supported list. Not so with 72h.

Also comment function 👍

Reply:
time.ParseDuration doesn't support "d" as time unit 
valid units are 
"ns" 
"us" / "µs"
"ms"
"s"
"m"
"h"

Just a user friendly way to still use time.ParseDuration (didn't want to reinvent the wheel and have a huge switch case for each supported type) but with support for 1d 3d options

Reply:
Ahh, thanks for the explanation! So in that case, when chaning a script to use "1w", despite being a supported duration, crashes the script

Reply:
ah yes will fix that good pickup didn't test weekly

will add some script test coverage as well to make sure these pass in future if any further ones are added 

thank you